JustSaying/AwsTools/AwsClientFactoryProxy.cs

```diff
@@ -16,8 +16,13 @@ public AwsClientFactoryProxy(Func<IAwsClientFactory> awsClientFactoryFunc)
             _awsClientFactoryFunc = awsClientFactoryFunc;
         }
 
+        public AwsClientFactoryProxy(Lazy<IAwsClientFactory> factory)
```

Comment:
`() => factory.Value` is just a `Func<IAwsClientFactory`, will the other overload not suffice?

Reply:
Instead of:
```csharp
new AwsClientFactoryProxy(new Lazy<IAwsClientFactory>(ClientFactoryBuilder.Build))
```
you'd do:
```csharp
var lazyFactory = new Lazy<IAwsClientFactory>(ClientFactoryBuilder.Build);
new AwsClientFactoryProxy(() => lazyFactory.Value)
```
I can see how it would be easy to new it up inside the delegate (which would be wrong), or eagerly get the `.Value` here though. It just struck me as odd, but I see how it moves people towards "the pit of success".

Reply:
I can back that commit out, but I thought it was neater than doing it at the call site. I'm hoping if #431 goes somewhere we can get rid of `(I)AwsClientFactoryProxy` anyway.

Reply:
I think you should keep it as it solved a problem for you. Just throwing out an alternative:
```csharp
public AwsClientFactoryProxy(Func<IAwsClientFactory> factory, bool lazy = false)
{
    _awsClientFactoryFunc = lazy ? new Lazy(factory) : factory;
}
```